File: pptx-service/renderers/cta.py
from builder import add_bg, add_textbox, contrast_color
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.accent)
    action = data.get("newAction", "")
    text_color = contrast_color(t.accent)
    add_textbox(slide, 1.0, 2.2, 11.3, 4.0, action,
                t.font_display, 68, text_color, bold=True,
                align="center", wrap=True)

File: lib/pptx/slides/cta.ts
import type { CTASlide } from '@/lib/types';
import {
  py, SLIDE_W,
  setBackground, addTextBox,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: CTASlide, t: PptxTheme): void {
  setBackground(slide, '#' + t.accent);

  addTextBox(slide, data.newAction, {
    x: t.paddingIn, y: py(380), w: SLIDE_W - 2 * t.paddingIn,
    fontSize: Math.round(t.heroSize * 0.8), color: 'FFFFFF',
    fontFace: t.fontDisplay, bold: true, align: 'center', wrap: true,
  });
}
